### PR TITLE
[Bugfix] Fix unable to load some models

### PR DESCRIPTION
**Pull request by DarkLight1337**
This PR fixes more models broken by #10205 and/or #10304.

I have also added tests for check that each model can be initialized using dummy format.

### CODE REVIEW

**Review by youkaichao — APPROVED** [review]
thanks for the fix!

File: tests/models/test_initialization.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+@pytest.mark.parametrize("model_arch", HF_EXAMPLE_MODELS.get_supported_archs())
+def test_can_initialize(model_arch):
+    model_info = HF_EXAMPLE_MODELS.get_hf_info(model_arch)
+
+    if not model_info.is_available_online:
+        pytest.skip("Model is not available online")
+
+    try:
+        LLM(model_info.default,
```
Comment (youkaichao) [inline tests/models/test_initialization.py:16]: would some models be too large to initialize in the ci's L4 GPU?

Comment (DarkLight1337) [reply tests/models/test_initialization.py:16]: Yeah, I am trying to find a solution to this...

Comment (youkaichao) [reply tests/models/test_initialization.py:16]: maybe adds some hf config override to make the model smaller. i do this when i test large models.

Comment (DarkLight1337) [reply tests/models/test_initialization.py:16]: Let me try setting `{"num_layers": 1, "num_hidden_layers": 1}`

**Comment by DarkLight1337** [comment]
After overriding the HF config with

```
{
        "num_layers": 1,
        "num_hidden_layers": 1,
        "num_experts": 2,
        "num_experts_per_tok": 2,
        "num_local_experts": 2,
}
```

I can even load [Snowflake/snowflake-arctic-instruct](https://huggingface.co/Snowflake/snowflake-arctic-instruct) model (originally with 479B parameters) in a single A10.

**Comment by DarkLight1337** [comment]
The tests pass locally now!

**Pull request merged**